generate_constraints

```diff
@@ -16,6 +16,9 @@ CONSTRAINTS_FILE=constraints.txt
 # handwritten python_requirement_library target.
 "${PIP}" install \
   -r "${REQUIREMENTS_FILE}" \
-  -r <(./pants dependencies --type=3rdparty ::)
+  -r <(./pants dependencies :: |
+    xargs ./pants filter --target-type=python_requirement |
+    xargs ./pants peek |
+    jq -r '.[]["requirements"][]')
```

Comment:
I can't find documentation for this, but it seems like the `--type` flag to `./pants dependencies` has been removed. This is now the way that the documentation suggests you filter for third party requirements: https://www.pantsbuild.org/docs/python-third-party-dependencies#tip-set-up-a-virtual-environment-optional

One annoying thing is that it requires you have `jq` installed, but that's a pretty common package 🤷🏻‍♀️